packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -0,0 +1,297 @@
+  }
+
+  setPollMs (pollMs: number) {
+    this.pollMs = pollMs
+  }
+
+  setTimeTilBoostMs (timeTilBoostMs: number) {
+    this.timeTilBoostMs = timeTilBoostMs
+  }
+
+  setGasPriceMutliplier (multiplier: number) {
+    this.gasPriceMultiplier = multiplier
+  }
+
+  setMaxGasPriceGwei (maxGasPriceGwei: number) {
+    this.maxGasPriceGwei = BigNumber.from(maxGasPriceGwei)
+  }
+
+  start () {
+    this.startPoller()
+  }
+
+  async save () {
+    await this.store.updateItem(this.id, this.marshal())
+  }
+
+  marshal (): MarshalledItem {
+    return {
+      id: this.id,
+      createdAt: this.createdAt,
+      txHash: this.txHash,
+      from: this.from,
+      to: this.to,
+      data: this.data,
+      value: this.value?.toString(),
+      nonce: this.nonce,
+      gasPrice: this.gasPrice?.toString(),
+      gasLimit: this.gasLimit?.toString()
+    }
+  }
+
+  static async fromId (id: string, signer: Signer, store: Store) {
+    const item = await store.getItem(id)
+    return GasBoostTransaction.unmarshal(item, signer, store)
+  }
+
+  static async unmarshal (item: MarshalledItem, signer: Signer, store: Store) {
+    const tx = {
+      from: item.from,
+      to: item.to,
+      data: item.data,
+      value: item.value,
+      nonce: item.nonce,
+      gasPrice: item.gasPrice,
+      gasLimit: item.gasLimit
+    }
+    const gTx = new GasBoostTransaction(tx, signer, store)
+    gTx.id = item.id
+    gTx.createdAt = item.createdAt
+    gTx.txHash = item.txHash
+    return gTx
+  }
+
+  getQueueGroup (): string {
+    if (this.chainSlug) {
+      return this.chainSlug
+    }
+    const chainSlug = getProviderChainSlug(this.signer.provider)
+    if (!chainSlug) {
+      throw new Error('chain slug not found for contract provider')
+    }
+    this.chainSlug = chainSlug
+    return this.chainSlug
+  }
+
+  @queue
+  @rateLimitRetry
+  async send () {
+    const nonce = await this.signer.getTransactionCount('pending')
+    this.nonce = nonce
+    const tx = await this.signer.sendTransaction({
+      from: this.from,
+      to: this.to,
+      data: this.data,
+      value: this.value,
+      nonce: this.nonce,
+      gasPrice: this.gasPrice,
+      gasLimit: this.gasLimit
+    })
+
+    this.from = tx.from
+    this.to = tx.to
+    this.data = tx.data
+    this.value = tx.value
+    this.gasLimit = tx.gasLimit
+    this.gasPrice = tx.gasPrice
+
+    this.track(tx)
+    return this
+  }
+
+  async getBumpedGasPrice (): Promise<BigNumber> {
+    const marketGasPrice = await this.signer.getGasPrice()
+    const multiplier = this.gasPriceMultiplier
+    const bumpedGasPrice = getBumpedGasPrice(this.gasPrice, multiplier)
+    return marketGasPrice.lt(bumpedGasPrice) ? bumpedGasPrice : marketGasPrice
+  }
+
+  private handleConfirmation (txHash: string) {
+    this.txHash = txHash
+    this.clearInflightTxs()
+    const receipt = this.getReceipt(txHash)
+    this.emit(State.Confirmed, receipt)
+  }
+
+  private async getReceipt (txHash: string) {
```

Comment:
Opinion: If I understand this correctly, I would expect `getReceipt` to return a receipt immediately. If this is waiting for a tx to complete, I would expect this to be called `waitForReceipt`.

Similar to ethers.js `provider.getTransactionReceipt`: "Returns the transaction receipt for hash or null if the transaction has not been mined."